model-datastructure/src/commonMain/kotlin/org/modelix/model/LRUCache.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2024.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.modelix.model
+
+class LRUCache<K, V>(val maxSize: Int) {
```

Comment:
Can't we reuse something existing such as https://commons.apache.org/proper/commons-collections/apidocs/org/apache/commons/collections4/map/LRUMap.html

Reply:
This is multi platform code.

Reply:
https://github.com/MayakaApps/Kache could be an option there.